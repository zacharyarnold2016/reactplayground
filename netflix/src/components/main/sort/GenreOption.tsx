import React from "react";
import { setCurrentGenre, setFilms } from "../../../redux/films";
import { useDispatch } from "react-redux";
import { filterGenre } from "../../../redux/api";

const GenreOption = (props: any) => {
  const dispatch = useDispatch();
  const genre = props.genre.toString();
  const [trigger] = filterGenre(genre);

  const sortGenre = async () => {
    const payload = await trigger({ genre }).unwrap();

    dispatch(setCurrentGenre(genre));
    dispatch(setFilms(payload.data));
  };
  return <h4 onClick={sortGenre}>{genre}</h4>;
};

export default GenreOption;
