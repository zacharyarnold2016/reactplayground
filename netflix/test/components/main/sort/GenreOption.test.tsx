// @ts-nocheck
import { expect } from "@jest/globals";
import renderer from "react-test-renderer";
import GenreOption from "../../../../src/components/main/sort/GenreOption";
import { useNavigate, useParams } from "react-router";
import generateUrl from "../../../../src/helpers/generateUrlString";
import * as route from "react-router-dom";

jest.mock("react-router");
jest.mock("react-router-dom");
const spyFunction = jest.fn();
useNavigate = jest.fn(() => spyFunction);
useParams = jest.fn(() => {
  return { searchQuery: "test" };
});
generateUrl = jest.fn();

afterAll(() => {
  jest.unmock("react-router");
  jest.unmock("react-router-dom");
});
describe("GenreOption", () => {
  test("Snapshot", () => {
    route.useSearchParams.mockReturnValue([{ get: spyFunction }, spyFunction]);
    const component = renderer.create(<GenreOption genre="test" />);
    let tree = component.toJSON();
    expect(tree).toMatchSnapshot();
  });
  test("onClick", () => {
    route.useSearchParams.mockReturnValue([{ get: spyFunction }, spyFunction]);
    const component = renderer.create(<GenreOption genre="test" />);
    let tree = component.toJSON();
    tree.props.onClick();
    expect(generateUrl).toBeCalled();
  });
  test("isSelected", () => {
    route.useSearchParams.mockReturnValue([{ get: () => "test" }, spyFunction]);
    const component = renderer.create(<GenreOption genre="test" />);
    let tree = component.toJSON();
    tree.props.onClick();
    expect(generateUrl).toBeCalled();
  });
});
