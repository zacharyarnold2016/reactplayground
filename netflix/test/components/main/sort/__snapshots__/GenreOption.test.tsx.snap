// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`GenreOption Snapshot 1`] = `
<h4
  onClick={[Function]}
>
  test
</h4>
`;
